paasta_tools/cli/cmds/status.py

```diff
@@ -1075,26 +1101,59 @@ def print_flink_status(
         output.append(f"    No other information available in non-running state")
         return 0
 
-    output.append(
-        "    Jobs:"
-        f" {status['overview']['jobs-running']} running,"
-        f" {status['overview']['jobs-finished']} finished,"
-        f" {status['overview']['jobs-failed']} failed,"
-        f" {status['overview']['jobs-cancelled']} cancelled"
-    )
-    output.append(
-        "   "
-        f" {status['overview']['taskmanagers']} taskmanagers,"
-        f" {status['overview']['slots-available']}/{status['overview']['slots-total']} slots available"
-    )
+    if status["state"] == "running":
+        # Flink cluster overview from paasta api client
+        try:
+            overview = get_flink_overview_from_paasta_api_client(
+                service=service, instance=instance, client=client
+            )
+        except Exception as e:
+            output.append(PaastaColors.red(f"Exception when talking to the API:"))
+            output.append(str(e))
+            return 1
 
-    # Avoid cutting job name. As opposed to default hardcoded value of 32, we will use max length of job name
-    if status["jobs"]:
-        max_job_name_length = max(
-            [len(get_flink_job_name(job)) for job in status["jobs"]]
+        output.append(
+            "    Jobs:"
+            f" {overview.jobs_running} running,"
+            f" {overview.jobs_finished} finished,"
+            f" {overview.jobs_failed} failed,"
+            f" {overview.jobs_cancelled} cancelled"
         )
+        output.append(
+            "   "
+            f" {overview.taskmanagers} taskmanagers,"
+            f" {overview.slots_available}/{overview.slots_total} slots available"
+        )
+
+    flink_jobs = FlinkJobs()
+    flink_jobs.jobs = []
+    if status["state"] == "running":
+        # Flink cluster jobs from paasta api client
+        try:
+            flink_jobs = get_flink_jobs_from_paasta_api_client(
+                service=service, instance=instance, client=client
+            )
+        except Exception as e:
+            output.append(PaastaColors.red(f"Exception when talking to the API:"))
+            output.append(str(e))
+            return 1
+
+    jobs: List[FlinkJobDetails] = []
+    job_ids: List[str] = []
+    if flink_jobs.get("jobs"):
+        job_ids = [job.id for job in flink_jobs.get("jobs")]
+    try:
+        jobs = a_sync.block(get_flink_job_details, service, instance, job_ids, client)  # type: ignore
+    except Exception as e:
+        output.append(PaastaColors.red(f"Exception when talking to the API:"))
+        output.append(str(e))
```

Comment:
This was to avoid killing the whole command if we found partial failures, but now that you mention it it might be misleading to reply with a partial result. I think I'll just `return`. Thanks!